MekHQ/src/mekhq/campaign/parts/equipment/LargeCraftAmmoBin.java

```diff
@@ -270,9 +270,14 @@ public MissingPart getMissingPart() {
 
     @Override
     public void updateConditionFromEntity(boolean checkForDestruction) {
+        final String METHOD_NAME = "updateConditionFromEntity()"; //$NON-NLS-1$
         if(null != unit) {
             Mounted mounted = unit.getEntity().getEquipment(equipmentNum);
-            if(null != mounted) {
+            if (!(mounted.getType() instanceof AmmoType)) {
+                MekHQ.getLogger().log(LargeCraftAmmoBin.class, METHOD_NAME, LogLevel.WARNING,
+                        unit.getName() + " has invalid ammo bin at equipment number " + equipmentNum);
+            }
+            if(null != mounted && mounted.getType() instanceof AmmoType) {
```

Comment:
Mounted can't be null at this point because if it were it would have thrown a NullPointerException on line 276. If it's possible that this is null, then that needs to be part of the check (which should also be logged, since that's also a problem).

Reply:
Fixed.